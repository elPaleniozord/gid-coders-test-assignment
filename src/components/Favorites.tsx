import { useContext, useEffect, useState } from "react"
import { Beer, PunkAPIContext } from "../ctx/PunkAPIContext"
import BeerItem from "./BeerItem"

const Favorites = () => {
  const {favorites, } = useContext(PunkAPIContext)
  const [beerList, setBeerList] = useState<(Element | JSX.Element)[]>([])

  useEffect(() => {
    const stored = localStorage.getItem('favorite-beers')
    const json: Beer[] = stored ? JSON.parse(stored) : []

    setBeerList(json.map(beer => (<BeerItem key={`fav-${beer.id}`} {...beer} />)))

  }, [favorites])

  return (
    <div>
      <h2>Favorite Beers</h2>
      { beerList.length === 0 ? (<p>You don't have any favorites yet</p>) : <ul>{beerList}</ul> }
    </div>
  )
}

export default Favorites