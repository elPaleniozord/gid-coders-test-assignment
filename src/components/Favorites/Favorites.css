.favorites-container {
  border: 2px solid var(--offwhite);
  border-radius: 5px;
  margin: .4rem;
  margin-top: 180px;
  color: var(--offwhite);
  cursor: pointer;
}
.favorites-header {
  text-align: center;
  font-size: 2.2rem;
}
.favorites-icon {
  height: 30px;
  transition: all .4s ease-in-out;
  fill: var(--highlight);
}
.favorites-icon__flipped {
  transform: scaleY(-1);
}
.favorites-collapsible {
  transition: all .4s ease-in-out;
  max-height: 0;
  height: 100%;
  overflow: hidden;
  opacity: 0;
}
.favorites-collapsible__open {
  max-height: 1000px;
  opacity: 1;
}

.favorites-list {
  padding: 0;
  list-style: none;
  color: var(--offblack);
  overflow-y: auto;
  max-height: 100vh;
}