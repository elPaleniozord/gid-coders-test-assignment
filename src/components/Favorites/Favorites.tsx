import { MouseEventHandler, useContext, useEffect, useState } from "react"
import { Beer, PunkAPIContext } from "../../ctx/PunkAPIContext"
import BeerItem from "../BeerItem/BeerItem"
import './Favorites.css'

const Favorites = () => {
  const {favorites, } = useContext(PunkAPIContext)
  const [beerList, setBeerList] = useState<(Element | JSX.Element)[]>([])
  const [expanded, setCollapsed] = useState(false)

  useEffect(() => {
    const stored = localStorage.getItem('favorite-beers')
    const json: Beer[] = stored ? JSON.parse(stored) : []

    setBeerList(json.map(beer => (<BeerItem key={`fav-${beer.id}`} {...beer} />)))

  }, [favorites])

  const handleClick: MouseEventHandler = (e) => {
    e.stopPropagation()
    setCollapsed(!expanded)
  }

  return (
    <div className="favorites-container" onClick={handleClick}>
      <h2 className="favorites-header">Favorite Beers </h2>

      {expanded && (
        beerList.length === 0 ? (<p className="favorites-caption">You don't have any favorites yet</p>) : <ul className="favorites-list">{beerList}</ul>
      )}     
    </div>
  )
}

export default Favorites