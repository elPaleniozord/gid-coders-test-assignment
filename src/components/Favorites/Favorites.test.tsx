import { render } from '@testing-library/react';
import Favorites from "./Favorites"

test('should render Favorites Component properly', () => {
  const { container } = render(<Favorites />)
  const favorites = container.querySelector('div')
  expect(favorites).toMatchSnapshot()
})