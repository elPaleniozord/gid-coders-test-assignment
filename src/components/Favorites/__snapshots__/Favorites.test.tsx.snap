// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render Favorites Component properly 1`] = `
<div
  class="favorites-container"
>
  <h2
    class="favorites-header"
  >
    Favorite Beers
  </h2>
  <div
    class="favorites-collapsible "
  >
    <p
      class="favorites-caption"
    >
      You don't have any favorites yet
    </p>
  </div>
  <svg
    class="favorites-icon false"
    viewBox="0 0 20 20"
  >
    <path
      d="M19 6.25l-1.5-1.5-7.5 7.5-7.5-7.5L1 6.25l9 9 9-9z"
    />
  </svg>
</div>
`;
